XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Hands/BaseHand.cs

```diff
@@ -0,0 +1,134 @@
+                : base(trackingState, controllerHandedness, inputSource, interactions)
+        {
+        }
+
+        /// <inheritdoc />
+        public override MixedRealityInteractionMapping[] DefaultLeftHandedInteractions => DefaultInteractions;
+
+        /// <inheritdoc />
+        public override MixedRealityInteractionMapping[] DefaultRightHandedInteractions => DefaultInteractions;
+
+        public override void SetupDefaultInteractions(Handedness controllerHandedness)
+        {
+            AssignControllerMappings(DefaultInteractions);
+        }
+
+        #region Protected InputSource Helpers
+
+        #region Gesture Definitions
+
+        protected void UpdateVelocity()
+        {
+            if (frameOn == 0)
+            {
+                deltaTimeStart = Time.unscaledTime;
+                lastPosition = GetJointPosition(TrackedHandJoint.Palm);
+                lastPalmNormal = GetPalmNormal();
+                //lastPalmNormal = new Vector3(1, 0, 0);
+            }
+            else if (frameOn == velocityUpdateInterval)
+            {
+                //update linear velocity
+                float deltaTime = Time.unscaledTime - deltaTimeStart;
+                Vector3 newVelocity = (GetJointPosition(TrackedHandJoint.Palm) - lastPosition) / deltaTime;
+                Velocity = (Velocity * 0.8f) + (newVelocity * 0.2f);
+
+                //update angular velocity
+                Vector3 currentPalmNormal = GetPalmNormal();
+                //currentPalmNormal = new Vector3(0, 1, 0);
+                Quaternion rotation = Quaternion.FromToRotation(lastPalmNormal, currentPalmNormal);
+                Vector3 rotationRate = rotation.eulerAngles * Mathf.Deg2Rad;
+                AngularVelocity = rotationRate / deltaTime;
+            }
+
+            frameOn++;
+            frameOn = frameOn > velocityUpdateInterval ? 0 : frameOn;
+        }
+
+        #endregion Gesture Definitions
+
+        public abstract bool TryGetJoint(TrackedHandJoint joint, out MixedRealityPose pose);
+
+        private Vector3 GetJointPosition(TrackedHandJoint jointToGet)
+        {
+            if (TryGetJoint(jointToGet, out MixedRealityPose pose))
+            {
+                return pose.Position;
+            }
+            return Vector3.zero;
+        }
+
+        protected Vector3 GetPalmNormal()
+        {
+            if (TryGetJoint(TrackedHandJoint.Palm, out MixedRealityPose pose))
+            {
+                return -pose.Up;
+            }
+            return Vector3.zero;
+        }
+
+        private float DistanceSqrPointToLine(Vector3 lineStart, Vector3 lineEnd, Vector3 point)
+        {
+            if (lineStart == lineEnd)
+            {
+                return (point - lineStart).magnitude;
+            }
+
+            float lineSegmentMagnitude = (lineEnd - lineStart).magnitude;
+            Vector3 ray = (lineEnd - lineStart);
+            ray *= (1.0f / lineSegmentMagnitude);
```

Comment:
Couldn't this be simplified as:
`Vector3 ray = (lineEnd - lineStart).Normalize();`

Reply:
It can. Goot catch!

Reply:
But turns out `lineSegmentMagnitude` is used a few lines below again. So changing this to `.Normalize()` or `.normalized` would require computing the original vector's magnitude twice. So in Terms of Performance the current implementation is better.